test/metric_value_benchmark/ecs_daemon_base_test_runner.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric_value_benchmark
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/test"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+)
+
+type IECSTestRunner interface {
+	validate() status.TestGroupResult
+	getTestName() string
+	getAgentConfigFileName() string
+	getAgentRunDuration() time.Duration
+	getMeasuredMetrics() []string
+}
+
+type ECSBaseTestRunner struct {
+	MetricFetcherFactory *metric.MetricFetcherFactory
+}
+
+type IAgentRunStrategy interface {
+	runAgent(e *environment.MetaData, configFilePath string) error
+}
+
+type ECSAgentRunStrategy struct {
+}
+
+func (r *ECSAgentRunStrategy) runAgent(e *environment.MetaData, configFilePath string) error {
+	b, err := os.ReadFile(configFilePath)
+	if err != nil {
+		return fmt.Errorf("Failed while reading config file")
+	}
+
+	agentConfig := string(b)
+	ssmParamType := "String"
+
+	err = test.PutParameter(&(e.CwagentConfigSsmParamName), &agentConfig, &ssmParamType)
+	if err != nil {
+		return fmt.Errorf("Failed while reading config file : %s", err.Error())
+	}
+	log.Printf("Put parameter successful")
+
+	err = test.RestartDaemonService(&(e.EcsClusterArn), &(e.EcsServiceName))
```

Comment:
Really good point. Unclear to me. Couldn't find public docs about that. So far this hasn't failed to use the new config I'm putting, so I have to guess this might support strongly consistent read.